standalone/pruntime/enclave/src/system/master_key.rs

```diff
@@ -1,33 +1,45 @@
-use crate::std::io::{Read, Write};
+use crate::std::convert::TryInto;
 use crate::std::path::PathBuf;
 use crate::std::prelude::v1::*;
 use crate::std::sgxfs::SgxFile;
 
-use phala_crypto::sr25519::{Persistence, Signing, SECRET_KEY_LENGTH, SIGNATURE_BYTES};
+use phala_crypto::sr25519::{Persistence, Signature, Signing, Sr25519SecretKey};
 use sp_core::sr25519;
 
+use serde::{Deserialize, Serialize};
+use serde_bytes;
+
 /// Master key filepath
 pub const MASTER_KEY_FILE: &str = "master_key.seal";
 
+#[derive(Debug, Deserialize, Serialize)]
+struct PersistentMasterKey {
+    #[serde(with = "serde_bytes")]
+    secret_vec: Vec<u8>,
+    #[serde(with = "serde_bytes")]
+    signature_vec: Vec<u8>,
+}
+
 fn master_key_file_path(sealing_path: String) -> PathBuf {
     PathBuf::from(&sealing_path).join(MASTER_KEY_FILE)
 }
 
 /// Seal master key seed with signature to ensure integrity
 pub fn seal(sealing_path: String, master_key: &sr25519::Pair, identity_key: &sr25519::Pair) {
     let secret = master_key.dump_secret_key();
-    let sig = identity_key.sign_data(&secret);
-
-    // TODO(shelven): use serialization rather than manual concat.
-    let mut buf = Vec::new();
-    buf.extend_from_slice(&secret);
-    buf.extend_from_slice(sig.as_ref());
+    let signature_vec = identity_key.sign_data(&secret).0.to_vec();
+    let secret_vec = secret.to_vec();
 
+    let data = PersistentMasterKey {
```

Comment:
Is it compatible with the current testnet version?

Reply:
NO, this version should get online in our next testnet

Reply:
I'll leave a TODO in WorkerPubkey persistency to ensure it is also refactored in our next testnet